internal/pkg/cli/validate.go

```diff
@@ -134,3 +158,37 @@ func stringPortValidation(val string) error {
 	}
 	return nil
 }
+
+// s3 bucket names: 'a-zA-Z0-9.-'
+func s3BucketNameValidation(val interface{}) error {
+	s, ok := val.(string)
+	if !ok {
+		return errValueNotAString
+	}
+	if len(s) < 3 || len(s) > 63 {
```

Comment:
are these numbers defined anywhere? maybe we can move them to consts inside the function and add a comment with a link

Reply:
I also found that there are some more arcane S3 bucket formatting restrictions, which I've added in my latest commit (to disallow trailing dashes, consecutive dots or dashes, no bucket names formatted like IP addresses).